### PR TITLE
Now Requests Pronouns of New User

### PR DESCRIPTION
**Pull request by sampoder**
Added stuff:

- Clippy now asks What are your pronouns? (how you want to be referred to by others)

Screen Shot 2020-04-30 at 9 02 29 PM

- Clippy responds to those that choose Something Else with "Awesome! Could you please elaborate on how would you like us to address you in the format of They/Them/Theirs?"

- Once it has all the details it says "Awesome! Taken note of. Every profile here has a custom field for pronouns, I recommend you add yours there to let others know! Here is a quick tutorial." or  "Great! Taken note of. Every profile here has a custom field for pronouns, I recommend you add yours there to let others know! Here is a quick tutorial." The tutorial link works in slack links here: https://api.slack.com/reference/block-kit/block-elements

- When sending to welcome committee it uses their pronouns! Even those who chose something else (that's why we needed the format).

- It also says their pronouns in the welcome committee message

Screen Shot 2020-04-30 at 9 06 30 PM

- At the end if they choose She or They it links them to the #orpheus-legion channel

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -85,6 +201,64 @@ app.action('intro_progress', async ({ ack, body }) => {
...
+});
+
+app.action('something_else', async ({ ack, body }) => {
+  ack();
+  
+  await sendMessage(body.channel.id, `Awesome! Could you please elaborate on how would you like us to address you in the format of They/Them/Theirs?`)
+  
+});
+
+app.action('hs-question', async ({ ack, body }) => {
```
Comment (lachlanjc) [inline app.js:259]: This is the only hyphenated action name & it doesn’t do anything, just want to check that those are intentional?

File: app.js [hunk 2]
```diff
@@ -182,8 +407,12 @@ app.action('coc_acknowledge', async ({ ack, body }) => {
   await sendMessage(body.channel.id, shipDesc, 10, finalTs)
   await sendMessage(body.channel.id, codeDesc, 10, finalTs)
   await sendMessage(body.channel.id, `Here are a bunch of other active channels that you may be interested in:`, 10, finalTs)
-  await sendMessage(body.channel.id, `<#C0JDWKJVA> <#C0NP503L7> <#C6LHL48G2> <#C0DCUUH7E> <#CA3UH038Q> <#C90686D0T> <#CCW6Q86UF> <#C1C3K2RQV> <#CCW8U2LBC> <#CDLBHGUQN> <#CDJV1CXC2> <#C14D3AQTT> <#CBX54ACPJ> <#CC78UKWAC> <#C8P6DHA3W> <#C010SJJH1PT> <#CDJMS683D> <#CDN99BE9L> <#CSHEL6LP5>`, 10, finalTs)
-
+  if (pronouns === "They/them/theirs" || pronouns === "She/her/hers"){
+  await sendMessage(body.channel.id, `<#C0JDWKJVA> <#C0NP503L7> <#C6LHL48G2> <#C0DCUUH7E> <#CA3UH038Q> <#C90686D0T> <#CCW6Q86UF> <#C1C3K2RQV> <#CCW8U2LBC> <#CDLBHGUQN> <#CDJV1CXC2> <#C14D3AQTT> <#CBX54ACPJ> <#CC78UKWAC> <#C8P6DHA3W> <#C010SJJH1PT> <#CDJMS683D> <#CDN99BE9L> <#CSHEL6LP5> <#CFZMXJ3FB> (an awesome channel just for girls + nb people)`, 10, finalTs)
```
Comment (lachlanjc) [inline app.js:411]: I’m worried OL will blend in a little too much here. Does it feel weird if it’s a final message in that thread?

File: app.js (hunk 1 above)
Comment (sampoder) [reply app.js:259]: oh oops i left that in

File: app.js (hunk 2 above)
Comment (sampoder) [reply app.js:411]: ya, the problem was I didn't want to auto add them so the only place left was that message:

#hack-night #hackathons #gamedev #music #support #lgbtq #apple #design #android #cats #dogs #packages #photography #jenga #food #cooking #counttoamillion #film #playdungeon

Let me think of a better way to add it in

Comment (sampoder) [reply app.js]: I've updated it, what do you think of it now?

File: app.js [hunk 3]
```diff
@@ -407,11 +404,9 @@ app.action('coc_acknowledge', async ({ ack, body }) => {
...
   if (pronouns === "They/them/theirs" || pronouns === "She/her/hers"){
-  await sendMessage(body.channel.id, `<#C0JDWKJVA> <#C0NP503L7> <#C6LHL48G2> <#C0DCUUH7E> <#CA3UH038Q> <#C90686D0T> <#CCW6Q86UF> <#C1C3K2RQV> <#CCW8U2LBC> <#CDLBHGUQN> <#CDJV1CXC2> <#C14D3AQTT> <#CBX54ACPJ> <#CC78UKWAC> <#C8P6DHA3W> <#C010SJJH1PT> <#CDJMS683D> <#CDN99BE9L> <#CSHEL6LP5> <#CFZMXJ3FB> (an awesome channel just for girls + nb people)`, 10, finalTs)
-  }
-  else{
-   await sendMessage(body.channel.id, `<#C0JDWKJVA> <#C0NP503L7> <#C6LHL48G2> <#C0DCUUH7E> <#CA3UH038Q> <#C90686D0T> <#CCW6Q86UF> <#C1C3K2RQV> <#CCW8U2LBC> <#CDLBHGUQN> <#CDJV1CXC2> <#C14D3AQTT> <#CBX54ACPJ> <#CC78UKWAC> <#C8P6DHA3W> <#C010SJJH1PT> <#CDJMS683D> <#CDN99BE9L> <#CSHEL6LP5>`, 10, finalTs)
+    await sendMessage(body.channel.id, `Also I think you'll love <#CFZMXJ3FB>, it's a channel where many of the women/non-binary people in Hack Club are!
```
Comment (lachlanjc) [inline app.js:409]:
```suggestion
    await sendMessage(body.channel.id, `Also, check out <#CFZMXJ3FB>—it’s a channel for women/femme/non-binary people in Hack Club!`, 10, finalTs)
```

**Comment by MatthewStanciu** [comment]
I made a few small changes:

* The welcome committee message now looks like "@Name (pronouns) joined the Slack"
* Used Airtable to set and get pronouns to avoid errors when more than one person is going through Clippy at once, and to collect gender data
* A few completely unrelated changes to the copy while I'm at it

I'm going to go ahead and merge, but please let me know if you have any questions/suggestions about this. Thanks so much for adding this ❤️

**Comment by lachlanjc** [comment]
Great improvements! Thank you both <3

**Pull request merged**